www/review

```diff
@@ -26,6 +25,8 @@ $isOldReview = false;
 // get the game ID
 $id = $_REQUEST['id'];
 
+if ($_REQUEST['userid']) $userid = $_REQUEST['userid'];
+else $userid = $_SESSION['logged_in_as'];

```

Comment:
Wait a minute! This isn't OK. The `$_REQUEST` variable is under the user's control. They can type anything they want in the `&userid` parameter.

For example, in your branch, login as a non-admin user and go here: http://localhost:8080/review?id=ckdsz92q8cws7r7c&userid=1k0mok394bbghxdp

Expected: You're not Jacq, so you shouldn't be allowed to edit her review
Actual: You're not only allowed to see the "edit review" page, you can edit the review, save, and publish it, as the wrong user.

Reply:
I think it should be OK if we do this (but I haven't tested it):

```php
$userid = $_SESSION['logged_in_as'];
if ($_REQUEST['userid'] && check_admin_privileges($db, $userid)) {
    $userid = $_REQUEST['userid'];
}
```

Reply:
Updated the code.
Nice catch, thanks.